portal/server/controllers/utilisation-report-service/previous-reports/index.ts

```diff
@@ -0,0 +1,58 @@
+import { Request, Response } from 'express';
+import api from '../../../api';
+import { getMonthName } from '../../../helpers/getMonthName';
+import { PRIMARY_NAV_KEY } from '../../../constants';
+import { asLoggedInUserSession } from '../../../helpers/express-session';
+
+type GetPreviousReportsRequest = Request & {
+  query: {
+    targetYear?: string;
+  };
+};
+
+export const getPreviousReports = async (req: GetPreviousReportsRequest, res: Response) => {
```

Comment:
Our original previous reports controller has always been a bit of a pain point for me as the return type of `api.getPreviousUtilisationReports...` in both `portal-api` and `dtfs-central-api` is different to what we would expect. This felt like a good opportunity to refactor this controller to typescript, make it a bit more readable and add types to document what gets returned